src/plugins/lastfm.tsx

```diff
@@ -136,73 +175,112 @@ export default definePlugin({
     },
 
     async fetchTrackData(): Promise<TrackData | null> {
-        if (!this.settings.username || !this.settings.apiKey) return null;
+        if (!settings.store.username || !settings.store.apiKey)
+            return null;
 
-        const response = await fetch(`https://ws.audioscrobbler.com/2.0/?method=user.getrecenttracks&api_key=${this.settings.apiKey}&user=${this.settings.username}&limit=1&format=json`);
-        const trackData = (await response.json()).recenttracks.track[0];
+        try {
+            const params = new URLSearchParams({
+                method: "user.getrecenttracks",
+                api_key: settings.store.apiKey,
+                user: settings.store.username,
+                limit: "1",
+                format: "json"
+            });
 
-        if (!trackData["@attr"]?.nowplaying) return null;
+            const res = await fetch(`https://ws.audioscrobbler.com/2.0/?${params}`);
+            if (!res.ok) throw `${res.status} ${res.statusText}`;
 
-        // why does the json api have xml structure
-        return {
-            name: trackData.name || "Unknown",
-            album: trackData.album["#text"],
-            artist: trackData.artist["#text"] || "Unknown",
-            url: trackData.url,
-            imageUrl: (trackData.image || []).filter(x => x.size === "large")[0]?.["#text"]
-        };
+            const json = await res.json();
+            if (json.error) {
+                logger.error("Error from Last.fm API", `${json.error}: ${json.message}`);
```

Comment:
very small nitpick: you could just make this
```suggestion
                throw `${json.error}: ${json.message}`;
```
to have slightly less duplicated code (the catch already logs and returns null)

Reply:
This is an error from the API itself, rather than an error parsing the response.